include/anvill/Type.h

```diff
@@ -172,12 +233,8 @@ class TypeTranslator {
       llvm::Type *type,
       EncodingFormat alphanum = EncodingFormat::kDefault) const;
 
-  // Parse an encoded type string into its represented type.
-  //
-  // See `docs/TypeEncoding.md` for information on how different types are
-  // represented.
   Result<llvm::Type *, TypeSpecificationError>
-  DecodeFromString(const std::string_view str) const;
+  DecodeFromSpec(TypeSpec spec) const;
```

Comment:
I think we also need a `TypeSpec EncodeToSpec(llvm::Type *type);`. I'll work on that now.

Reply:
Are we sure we actually need it?

Reply:
Maybe not, looks like most of the calls are in the JSON generation part + unit tests. I'll try removing the JSON bits first.

Reply:
Oh probably not. Anyway, I hadn't written that yet so all good.